### PR TITLE
Clean-up of deprecated methods and interfaces

### LINKED ISSUE #20877: Clean-up of deprecated methods and interfaces
In previous releases of Keycloak there was a clean-up of provider and model interfaces
where some methods were deprecated. This issue aims to remove some of the deprecations.

### PR DESCRIPTION
**Pull request by vramik**
- `RealmModel#searchForGroupByNameStream`  - deprecated since 20.0.0
- `UserProvider#getUsersStream(RealmModel, boolean)` - deprecated since 20.0.0
- `UserSessionPersisterProvider#loadUserSessions` - deprecated since 13.0.0
- `Streams` interfaces - deprecated since 20.0.0
- `KeycloakModelUtils#getClientScopeMappings` - deprecated since 12.0.0
- `KeycloakSession` - deprecated since 19.0.0

Fixes #20877

### CODE REVIEW

**Comment by hmlnarik** [comment]
I like the direction where this PR is going, and the removals make sense to me.

I'm +1 for including this for 22 since it simplifies the interfaces. Any thoughts, @mposolda @pedroigor @stianst ?

**Comment by mposolda** [comment]
To me, it also looks good and definitely +1 to include in Keycloak 22. More legacy stuff removed, the better :-)

One quick though I've noticed: It seems that method `LegacySessionSupportProvider.userCache()` can be removed as well (at least it seems to not be used anywhere with this PR changes and I guess it it not supposed to be used directly by user's application?). But that's just a detail.

**Comment by vramik** [comment]
> One quick though I've noticed: It seems that method `LegacySessionSupportProvider.userCache()` can be removed as well (at least it seems to not be used anywhere with this PR changes and I guess it it not supposed to be used directly by user's application?).

Thanks @mposolda for pointing this out. I've incorporated it into the PR.

**Comment by hmlnarik** [comment]
@andymunro Could you please review the docs in this PR?

File: model/legacy/src/main/java/org/keycloak/storage/adapter/AbstractUserAdapterFederatedStorage.java [hunk 1]
```diff
@@ -459,101 +418,14 @@ public int hashCode() {
     }
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link AbstractUserAdapterFederatedStorage} default by providing
-     * implementations that delegate to the {@link Stream}-based variants instead of the other way around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
+     * and therefore the parent interface can be used directly
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/adapter/AbstractUserAdapterFederatedStorage.java:422]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
     * and therefore the parent interface can be used directly
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserAttributeFederatedStorage.java [hunk 2]
```diff
@@ -47,27 +40,13 @@ public interface UserAttributeFederatedStorage {
...
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link UserAttributeFederatedStorage}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
+     * and therefore the parent interface can be used directly
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserAttributeFederatedStorage.java:47]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
     * and therefore the parent interface can be used directly
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserBrokerLinkFederatedStorage.java [hunk 3]
```diff
@@ -35,41 +33,22 @@ public interface UserBrokerLinkFederatedStorage {
...
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link UserBrokerLinkFederatedStorage}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
+     * and therefore the parent interface can be used directly
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserBrokerLinkFederatedStorage.java:49]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
     * and therefore the parent interface can be used directly
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserFederatedStorageProvider.java [hunk 4]
```diff
@@ -83,13 +72,10 @@ default Stream<String> getStoredUsersStream(RealmModel realm, Integer first, Int
     void preRemove(RealmModel realm, ComponentModel model);
 
     /**
-     * The {@link UserFederatedStorageProvider.Streams} interface makes all collection-based methods in {@link UserFederatedStorageProvider}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserFederatedStorageProvider.java:75]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserFederatedUserCredentialStore.java [hunk 5]
```diff
@@ -66,36 +49,15 @@ default Stream<CredentialModel> getStoredCredentialsStream(RealmModel realm, Str
...
     CredentialModel getStoredCredentialByNameAndType(RealmModel realm, String userId, String name, String type);
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link UserFederatedUserCredentialStore}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserFederatedUserCredentialStore.java:57]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserRequiredActionsFederatedStorage.java [hunk 6]
```diff
@@ -28,42 +26,23 @@
...
     void removeRequiredAction(RealmModel realm, String userId, String action);
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link UserRequiredActionsFederatedStorage}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserRequiredActionsFederatedStorage.java:42]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
```

File: model/legacy/src/main/java/org/keycloak/storage/federated/UserRoleMappingsFederatedStorage.java [hunk 7]
```diff
@@ -44,28 +36,15 @@ public interface UserRoleMappingsFederatedStorage {
...
     void deleteRoleMapping(RealmModel realm, String userId, RoleModel role);
 
     /**
-     * The {@link Streams} interface makes all collection-based methods in {@link UserRoleMappingsFederatedStorage}
-     * default by providing implementations that delegate to the {@link Stream}-based variants instead of the other way
-     * around.
-     * <p/>
-     * It allows for implementations to focus on the {@link Stream}-based approach for processing sets of data and benefit
-     * from the potential memory and performance optimizations of that approach.
+     * @deprecated This interface is no longer necessary, collection-based methods were removed from the parent interface
```
Comment (andymunro) [inline model/legacy/src/main/java/org/keycloak/storage/federated/UserRoleMappingsFederatedStorage.java:44]:
```suggestion
     * @deprecated This interface is no longer necessary; collection-based methods were removed from the parent interface
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, @hmlnarik. Just a few changes.

**Review by hmlnarik — DISMISSED** [review]
Thank you for the PR!

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**